### PR TITLE
[WIP] Documentation style guide

### PR DESCRIPTION
**Pull request by mrbobbytables**
First pass a documentation style-guide for the contributor documentation in the community repo (related issue: #2850).

### CODE REVIEW

**Comment by parispittman** [comment]
Thanks for the work, @mrbobbytables!! 💯

**Comment by cblecker** [comment]
This is marked WIP, but when it's ready for review, please let me know. I'd love to help review it.

/cc

File: contributors/guide/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+* Format dates as `month day, year`. (December 13, 2018)
+  * **Good example:** October 24, 2018
+  * **Bad example:** 10/24/18
+* When conveying a date in numerical form, use [ISO 8601] Format: `yyyy-mm-dd`.
+  * **Good example:** 2018-10-24
+  * **Bad example:** 10/24/18
+* Use the 24 hour clock when referencing time.
+  * **Good example:** 15:30
+  * **Bad example:** 3:30pm
+* Times should be expressed in the UTC timezone in addition to any local times.
```
Comment (coderanger) [inline contributors/guide/style-guide.md]: Had a bit of a bikeshed in meeting, just noting it so it doesn't get lost, should figure out recommendations for recurring events re: timezones.

Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: It could be updated to be something like "Time should be expressed in a daylight savings  agnostic time such as ET or PT." and require linkout to the timezone converter service.

Comment (spiffxp) [reply contributors/guide/style-guide.md]: Roughly restating what I heard Noah say during the meeting:
- single point in time events should be expressed in an absolute timezone (e.g. use UTC instead of GMT, use EDT instead of ET)
- recurring events should be expressed in the most appropriate local timezone that follows DST changes (e.g. use GMT instead of UTC, or use PT instead of PST/PDT)

Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: I like that idea. Should cover all our use-cases.

File: contributors/guide/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+  * [Tables](#tables)
+* [Attribution](#attribution)
+
+
+## Cheatsheet
+
+### Cheatsheet: Content design, formatting, and language
+
+**[Contact information:](#contact-information)**
+* Use official Kubernetes contact information.
```
Comment (idvoretskyi) [inline contributors/guide/style-guide.md]: > * Use official Kubernetes contact information.

What does it mean (clarification needed)?

File: contributors/guide/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+    * **Bad example:** The flag --advertise-address is used to denote the IP 
+      address on which to advertise the apiserver to members of the cluster.
+  * Object field names.
+    * **Good example:** Set the `externalTrafficPolicy` to Local.
+    * **Bad example:** Set the externalTrafficPolicy to Local.
+
+
+### Language, grammar and tone
+
+* Documentation should be written in US English.
```
Comment (idvoretskyi) [inline contributors/guide/style-guide.md]: Why US English is mandatory (not just preferred)?

File: contributors/guide/style-guide.md (hunk 2 above)
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: It's expanded on further down. The top portion is intended to just be  quick reference of the guidelines. The full block is this:
* Use official Kubernetes contact information.
   * Use official community contact email addresses. There should be no personal or work 
      contact information included in public documentation; instead use addresses like the SIG 
      Google groups or managed accounts such as community@kubernetes.io.
    * **Good example:** community@kubernetes.io
    * **Bad example:** bob@example.com

File: contributors/guide/style-guide.md [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+
+* When moving a document, leave a tombstone file with a removal date in it's
+  place.
+  * Tombstones function as a pointer and give users a time to update their own
+    documentation and bookmarks. Their usefulness is time-bounded and should be
+    removed when they would logically no longer serve their purpose.
+    ```markdown
+    This file has moved to https://git.k8s.io/community/contributors/guide/README.md.
+
+    This file is a placeholder to preserve links.  Please remove after 3 months or the release of kubernetes 1.10, whichever comes first.
```
Comment (eduartua) [inline contributors/guide/style-guide.md]: Could we use an exact date in the format suggested at Dates and Time (e.g. December 13, 2018) instead of 3 months?

Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: updated with explicit date in the example :)

File: contributors/guide/style-guide.md [hunk 5]
```diff
@@ -0,0 +1,672 @@
...
+* Use a colon (`:`) to separate a list item name from the explanatory text.
+* Leave a blank line after each list.
+* Use `*` for unordered lists.
+* For ordered lists repeating `1.` may be used.
+* When inserting a code block into an ordered list, indent (space) an additional
+  two times.
+
+**[Metadata:](metadata)**
+* Include a yaml metadata header at the beginning of the document.
+* Metadata must include the `title` attribute.
```
[the remark this answers is not in the document]
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: The metadata is intended for use with the contributor site. Maybe rephrase to something like "If the document is to be surfaced on the contributor-site, include a metadata header"?

File: contributors/guide/style-guide.md [hunk 6]
```diff
@@ -0,0 +1,672 @@
...
+  * git.k8s.io -> github.com/kubernetes
+  * sigs.k8s.io -> github.com/kubernetes-sigs
+
+**[Lists:](#lists)**
+* Capitalize the first character of each entry unless the item is explicitly
+  case sensitive.
+* End each entry with a period if it is a sentence or phrase.
+* Use a colon (`:`) to separate a list item name from the explanatory text.
+* Leave a blank line after each list.
+* Use `*` for unordered lists.
```
[the remark this answers is not in the document]
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: The `*` decision was somewhat arbitrary XD It was pretty close to 50/50 when I was looking at docs. Really should just need to use one or the other lol.

**Comment by mrbobbytables** [comment]
> contribex wanted my hot takes
> 
> also I'm assuming some of the WIP is deduping between cheatsheet and the longer formatting guide below? some parts seem like copypasta

The guidelines themselves are copypasta. Idea was to have a quick reference without the examples/explanation. Realistically though it's probably short enough where it's not needed.

File: contributors/guide/style-guide.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (spiffxp) [reply contributors/guide/style-guide.md]: I have a personal preference for `-` because it doesn't require that I type the shift key.  This may be an emacs/vim style divide, no idea

File: contributors/guide/style-guide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (spiffxp) [reply contributors/guide/style-guide.md]: I mean, if we think everything here should have a metadata header because it's all in some way destined for that site, that's fine too.  This just raised my eyebrows as a new ideal state rather than something that describes how we mostly do things

[the remark this answers is not in the document]
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: I'd say flip to conditional for now. If we wind up in the situation where most of the content is being surfaced, the guideline can be updated to become the default.

File: contributors/guide/style-guide.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: sgtm. I had no real opinion one way or the other outside of just one being consistently used.

File: contributors/guide/style-guide.md [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+- Use official Kubernetes contact information.
+
+**[Dates and times:](#dates-and-times)**
+- Format dates as `month day, year`. (December 13, 2018)
+- When conveying a date in numerical form, use [ISO 8601] Format: `yyyy-mm-dd`.
+- Use the 24 hour clock when referencing time.
+- Times for single events (example: KubeCon) should be expressed in an absolute
+  timezone such as Pacific Standard Time (PST) or Coordinated Universal Time
+  (UTC).
+- Times for reoccurring events should be expressed in the PT (Pacific Time)
```
Comment (cblecker) [inline contributors/guide/style-guide.md]: - If it is the intent that a recurring event will follow Daylight Savings Time (DST), then the time should be expressed with a timezone that follows DST. This way, recurring events will not need to be updated twice a year.

File: contributors/guide/style-guide.md [hunk 8]
```diff
@@ -0,0 +1,683 @@
...
+  timezone or the most appropriate timezone that adheres to Daylight Savings
+  Time (DST).
+- Supply a link to a globally available time zone converter service.
+  - `http://www.thetimezoneconverter.com/?t=<TIME REFERENCE>&tz=PT%20%28Pacific%20Time%29`
```
Comment (cblecker) [inline contributors/guide/style-guide.md]: Add e.g. to mark this as an example (you're explicitly using PT)

File: contributors/guide/style-guide.md [hunk 9]
```diff
@@ -0,0 +1,683 @@
...
+  - Title in `H1` (a single `#`).
+  - A brief description or summary of the document.
+  - A table of contents.
+  - The general body of document.
+- Do not repeat content. Instead link back to the canonical source.
+- Large content or topic shifts should be separated with a horizontal rule.
+
+**[Formatting text:](#formatting-text)**
+- API objects:
+  - Use [camel case], or the format commonly used by default for the language.
```
Comment (cblecker) [inline contributors/guide/style-guide.md]: API standards are very specific things, and should probably be out of scope for this (the sig-arch api-approvers team is responsible for those). I'd perhaps link out to the API conventions: `/contributors/devel/sig-architecture/api-conventions.md`

File: contributors/guide/style-guide.md [hunk 10]
```diff
@@ -0,0 +1,683 @@
...
+**[Language, grammar and tone:](#language)**
+- Documentation should be written in US English.
+- Prefer an active voice and present tense when possible.
+- Use simple and direct language.
+- Use gender-neutral language.
+- Avoid personal pronouns ("I," "we," "us," "our," and "ours").
+- Address the reader as "you" instead of "we".
+- Do not use Latin phrases.
+- Avoid jargon and idioms.
+- If using acronyms, ensure they are clearly defined in the same document.
```
Comment (cblecker) [inline contributors/guide/style-guide.md]: or linked to another doc where defined

File: contributors/guide/style-guide.md [hunk 11]
```diff
@@ -0,0 +1,683 @@
...
+
+**[Moving a document:](#moving-a-document)**
+- When moving a document, leave a tombstone file with a removal date in it's
```
Comment (cblecker) [inline contributors/guide/style-guide.md]: Add a new bullet to `git mv` files in a separate commit wherever possible, as this will retain git history for the moved file.

File: contributors/guide/style-guide.md (hunk 8 above)
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: Swapped it to `<TZ REFERENCE>` and used the PT in the full example:
  ```
  The weekly SIG meeting is at [13:30 PT].

  [13:30 PT]: http://www.thetimezoneconverter.com/?t=13:30&tz=PT%20%28Pacific%20Time%29
  ```

File: contributors/guide/style-guide.md (hunk 7 above)
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: Updated to read: "Times for reoccurring events should be expressed in a time zone that follows Daylight Savings Time (DST) such as Pacific Time (PT) or Eastern Time (ET)."

The explanation regarding why (not having to update the doc) is expanded upon in the time formatting section.

Although,  I am unsure of how worthwhile it is to have the "cheatsheet" portion at the top and the full thing further down as the explanations and examples add more context.

File: contributors/guide/style-guide.md (hunk 9 above)
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: That doc itself is definitely the canonical source of truth, but can be pretty intimidating. Do you think linking to the [naming conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#naming-conventions) section directly would  be a good middle ground? It's sort of a "TL;DR" and the focus of this doc is more on the text formatting itself.

**Comment by cblecker** [comment]
All my comments appear to be addressed :)

File: contributors/guide/style-guide.md (hunk 11 above)
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: done :)

File: contributors/guide/style-guide.md (hunk 9 above)
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: Updated to point towards the api-conventions doc.

File: contributors/guide/style-guide.md [hunk 12]
```diff
@@ -0,0 +1,672 @@
...
+* Use official Kubernetes contact information.
+
+**[Dates and times:](#dates-and-times)**
+* Format dates as `month day, year`. (December 13, 2018)
+* When conveying a date in numerical form, use [ISO 8601] Format: `yyyy-mm-dd`.
+* Use the 24 hour clock when referencing time.
+* Times for single events (example: KubeCon) should be expressed in an absolute
+  timezone such as Pacific Standard Time (PST) or Coordinated Universal Time
+  (UTC).
+* Times for reoccurring events should be expressed in the PT (Pacific Time)
```
[the remark this answers is not in the document]
Comment (mrbobbytables) [reply contributors/guide/style-guide.md]: Rephrased to be a bit more generic:
"Times for reoccurring events should be expressed in a time zone that follows Daylight Savings Time (DST) such as Pacific Time (PT) or Eastern Time (ET)."

**Comment by mrbobbytables** [comment]
I think all the comments and suggestions have been addressed. Removing WIP for last review.

**Comment by cblecker** [comment]
/approve
Shipping it, per ContribEx 🚀
